Guides for OCI and AWS Metrics (#1042)
micronaut-projects/micronaut-guides: guides/micronaut-metrics-aws/micronaut-metrics-aws.adoc

external:micronaut-metrics/start.adoc[]

> common:aws.adoc[]

Reviewer: FYI: If you want the What you will need section to look the same as OCI see https://github.com/micronaut-projects/micronaut-guides/blob/master/guides/micronaut-cloud-database-aws/micronaut-cloud-database-aws.adoc

> common:requirements.adoc[]

Reviewer 2: remove this, it's in micronaut-metrics/start.adoc and getting rendered twice

external:micronaut-metrics/create-app.adoc[]

> == Monitor metrics on the AWS Cloudwatch

Reviewer 2: remove "the"

> For publishing metrics on the AWS Cloudwatch you have to adjust your application configuration slightly. With that configuration change you should be able to create queries and monitor your metrics on the https://console.aws.amazon.com/cloudwatch#metricsV2:graph=~()[Cloudwatch].

Reviewer 2: ```suggestion
To publish metrics to AWS Cloudwatch, you have to adjust your application configuration slightly, and then you will be able to create queries and monitor your metrics on https://console.aws.amazon.com/cloudwatch#metricsV2:graph=~()[Cloudwatch].
```

resource:application.yml[tag=metrics]

> <1> Change to `false` to disable all metrics, e.g., per-environment

Reviewer 3: you could extract a callout to avoid repetition between both guides.

> <2> These default to `true` and are only here for convenience, to be able to disable a subset of metrics

Reviewer 3: ```suggestion
<2> These binders default to `true`. We list them for convenience to encourage you to experiment with disabling some of them. 
```

> <3> Change namespace to something meaningful, for an example app name. In this guide we are using `micronaut_guide`

Reviewer 3: ```suggestion
<3> Change the namespace to something meaningful, for example, the app's name. In this guide, we are using `micronautguide`
```

We alway use `micronautguide` wtithout underscore for the guide application name. Is there a reason why you used `micronaut_guide`?

=== Visit the https://console.aws.amazon.com/cloudwatch#metricsV2:graph=~()[AWS Cloudwatch] Metrics.

> In the Metrics panel select your namespace that you chose earlier. We will select `micronaut_guide`.

Reviewer 3: ```suggestion
In the Metrics panel, select the namespace that you chose earlier. We select `micronautguide`.
```

We alway use micronautguide wtithout underscore for the guide application name. Is there a reason why you used `micronaut_guide`?

Author: There is none

> In the Metrics panel select your namespace that you choose earlier. We will select `micronaut_guide`.

Reviewer 2: choose -> chose

{empty} +

image::aws-metrics/cloudwatch-metrics.png[]

{empty} +

> Here you can browse different metrics and explore their values. In this example we want to check the price of Bitcoin. Select `Metrics with no dimensions`.

Reviewer 3: ```suggestion
Here you can browse different metrics and explore their values. In this example, we want to check the price of Bitcoin. Select `Metrics with no dimensions`.
```

> Here you can browse different metrics and explore their values. In this example we want to check the price of bitcoin. We will select `Metrics with no dimensions`.

Reviewer 2: ```suggestion
Here you can browse different metrics and explore their values. In this example we want to check the price of Bitcoin. Select `Metrics with no dimensions`.
```

{empty} +

image::aws-metrics/cloudwatch-metrics-browse.png[]

{empty} +

> In this panel you can see our custom metrics about Bitcoin. Select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how the price of Bitcoin changed over time.

Reviewer 3: ```suggestion
In this panel, you can see our custom metrics about Bitcoin. Select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how the price of Bitcoin changed over time.
```

> In this panel you can see our custom metrics about bitcoin. We can select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how price of the bitcoin were changing through time.

Reviewer 2: ```suggestion
In this panel you can see our custom metrics about Bitcoin. Select (#2) `bitcoin.price.latest.value` to see on (#1) the graph how the price of Bitcoin changed over time.
```
